docs/source/routing.rst

```diff
@@ -0,0 +1,240 @@
+============
+ Routing
+============
+
+In the quickstart tutorial, we saw how to create and render widgets,
+and how to react to user events. We also saw that the name of the app
+defined its base url: the ``tasks`` app was accessible under
+``localhost:8080/tasks/``.
+
+Here, we will extend the example and make each task accessible under
+``/tasks/<task id>``.
+
+Before we start, here's the summary on how to handle routing in
+Weblocks:
+
+* use ``weblocks-navigation-widget:defroutes``
+
+* ``defroutes`` associates urls (as strings) to a widget::
```

Comment:
@vindarel I would say that `defroutes` associates URLs to a form that generally evaluates to a widget. Another case could be a redirection to a different URL, for example.

Reply:
Hello, thanks for the review!

I am tempted to write:

> * ``defroutes`` associates URLs (as strings) to a widget (or more precisely, to a form that usually evaluates to a widget)

but this is too vague IMO.

Redirections are done with the function `(weblocks/response:redirect)`. I'll mention them.

Are there other cases where the thing associated to the URL would not be a widget, in order to be exhaustive?

Reply:
Any functionality that needs complete control over what is returned could be considered. For example, a REST API returning a json payload. But I think that, specially for a beginner reading a tutorial, it's a good thing to stress that you almost always want to associate a URL to a widget. So maybe you can put a note in there, but leaving your original description as is.